src/mito2/src/sst/file.rs

```diff
@@ -95,6 +95,23 @@ pub struct FileMeta {
     pub level: Level,
     /// Size of the file.
     pub file_size: u64,
+    /// Available indexes of the file.
+    pub available_indexes: Vec<IndexType>,
```

Comment:
Do we need to use smallvec to eliminate a possible heap allocation? Since we only have one index type.

What's more, I don't know how to store other indices in the future since we might use different index types for different columns. If we store them in the same file (since puffin supports this) then we might only need a flag to indicate whether there is an index file.

Reply:
> store them in the same file

I cannot predict the future, but I hope this holds true, and the `index_file_size` I added also suggests this.

> we might only need a flag to indicate whether there is an index file

Although a single file can represent various indices, a flag is sufficient to simply determine whether an index file exists. However, indicating which indices are available here is still valuable. As metadata of the index, users can use this information to make some valuable judgments without having to parse the file.

Reply:
> Do we need to use smallvec to eliminate a possible heap allocation?

Heap allocation is evil and I have addressed it.